src/h5web/vis-packs/core/shared/Html.tsx

```diff
@@ -4,63 +4,104 @@ import {
   forwardRef,
   HTMLAttributes,
   useLayoutEffect,
+  useRef,
 } from 'react';
 import ReactDOM from 'react-dom';
-import { useThree } from '@react-three/fiber';
+import { GroupProps, useFrame, useThree } from '@react-three/fiber';
+import { Group, Vector3 } from 'three';
 
 // Simplified version of `drei`'s `<Html>` component
-// https://github.com/pmndrs/drei/blob/v2.2.3/src/Html.tsx
-const Html = forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(
-  (props, ref) => {
-    const { className, style, children, ...divProps } = props;
+// https://github.com/pmndrs/drei/blob/v6.0.3/src/web/Html.tsx
 
-    const { width, height } = useThree((state) => state.size);
-    const gl = useThree((state) => state.gl);
-    const { parentElement } = gl.domElement;
+const v1 = new Vector3();
 
-    // Container `div` for ReactDOM to render into, appended next to R3F's `canvas`
-    const [el] = useState(() => document.createElement('div'));
+export interface HtmlProps extends HTMLAttributes<HTMLDivElement> {
+  groupProps?: GroupProps;
+  followCamera?: boolean;
+}
 
-    useEffect(() => {
-      el.style.cssText = `position: absolute; top: 0; left: 0;`;
+const Html = forwardRef<HTMLDivElement, HtmlProps>((props, ref) => {
+  const {
+    className,
+    style,
+    children,
+    followCamera,
+    groupProps = {},
+    ...divProps
+  } = props;
 
+  const { width, height } = useThree((state) => state.size);
+  const gl = useThree((state) => state.gl);
+  const { parentElement } = gl.domElement;
+  const camera = useThree((state) => state.camera);
+  const scene = useThree((state) => state.scene);
+
+  // Container `div` for ReactDOM to render into, appended next to R3F's `canvas`
+  const [el] = useState(() => document.createElement('div'));
+  // Placeholder R3F group in the canvas that will be the counterpart of HTML in the canvas.
+  // Moving the camera changes the group position. These position changes are forwarded to the HTML in updateHtmlPosition.
+  const group = useRef<Group>(null!); // eslint-disable-line @typescript-eslint/no-non-null-assertion
+
+  function updateHtmlPosition() {
+    camera.updateMatrixWorld();
+    const objectPos = v1.setFromMatrixPosition(group.current.matrixWorld);
+    objectPos.project(camera);
+    const widthHalf = width / 2;
+    const heightHalf = height / 2;
+    const pos = [
+      objectPos.x * widthHalf + widthHalf,
+      -(objectPos.y * heightHalf) + heightHalf,
+    ];
+
+    el.style.transform = `translate3d(${pos[0]}px,${pos[1]}px,0)`;
+  }
+
+  useEffect(() => {
```

Comment:
Found a small bug: when React hot-reloads (but probably also when `parentElement` changes, if ever), the clean up function of this `useEffect` runs _after_ the `useLayoutEffect` below, which means that `ReactDOM.unmountComponentAtNode(el);` is called _after_ `ReactDOM.render()`, which means that the annotation disappears from the screen.

Replacing this `useEffect` with a `useLayoutEffect` fixes the issue.